include/circt/Dialect/Seq/Seq.td

```diff
@@ -61,4 +61,47 @@ def CompRegOp : SeqOp<"compreg",
   ];
 }
 
+def FirRegOp : SeqOp<"firreg",
+    [NoSideEffect, AllTypesMatch<["input", "data"/*, "resetValue"*/]>,
+     SameVariadicOperandSize,
+     DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>]> {
+      // AllTypesMatch doesn't work with Optional types yet.
+
+  let summary = "Register with preset and sync or async reset";
+  let description = [{
+    See the Seq dialect rationale for a longer description.
+
+    ```
+    %name = seq.firreg %input, %clk [ sym @sym]
+        [ reset (sync|async) %reset, %value ] : type
+    ```
+  }];
+
+  let arguments = (ins AnyType:$input, I1:$clk, StrAttr:$name,
+                       OptionalAttr<SymbolNameAttr>:$inner_sym,
```

Comment:
Likely needs updating if this is operating in the firrtl realm.  if in the HW realm, this is fine for now.